libs/platform/src/lib/table/directives/table-cell-selectable.directive.ts

```diff
@@ -1,19 +1,106 @@
-import { Directive, HostListener } from '@angular/core';
-import { SPACE } from '@angular/cdk/keycodes';
+import {
+    Directive,
+    ElementRef,
+    HostBinding,
+    HostListener,
+    Inject,
+    Input,
+    OnChanges,
+    SimpleChanges
+} from '@angular/core';
+import { DOWN_ARROW, ENTER, F2, LEFT_ARROW, MAC_ENTER, RIGHT_ARROW, UP_ARROW } from '@angular/cdk/keycodes';
 
 import { KeyUtil } from '@fundamental-ngx/core/utils';
 
+import { TableCellNavigationId, TableService } from '../table.service';
+import { FIRST_CELL_NAVIGATION_ID } from '../constants';
+import { DOCUMENT } from '@angular/common';
+
+/** @dynamic */
 @Directive({
     // eslint-disable-next-line @angular-eslint/directive-selector
     selector: '[fdTableCellSelectable], [fd-table-cell-selectable]'
 })
-export class FdpCellSelectableDirective {
+export class FdpCellSelectableDirective implements OnChanges {
+    /** Used to register cell as focusable to have ability to navigate over it */
+    @Input('fd-table-cell-selectable')
+    navigationId: TableCellNavigationId;
+
+    /** @hidden */
+    @HostBinding('attr.tabindex')
+    _tabindex = -1;
+
+    /** @hidden */
+    constructor(
+        readonly _elRef: ElementRef,
+        private readonly _tableService: TableService,
+        @Inject(DOCUMENT) private readonly _document: Document | null
+    ) {}
+
+    /** @hidden */
+    ngOnChanges(changes: SimpleChanges): void {
+        if ('navigationId' in changes) {
+            this._tabindex = this.navigationId === FIRST_CELL_NAVIGATION_ID ? 0 : -1;
+            this._tableService.registerFocusableTableCell(this.navigationId, this);
+        }
+    }
+
     /** @hidden */
     @HostListener('keydown', ['$event'])
-    onKeyDown(event: KeyboardEvent): void {
-        if (KeyUtil.isKeyCode(event, SPACE)) {
-            // prevent page scrolling on Space keydown at checkbox/radio
+    _onKeyDown(event: KeyboardEvent): void {
+        if (
+            KeyUtil.isKeyCode(event, F2) ||
+            (KeyUtil.isKeyCode(event, [ENTER, MAC_ENTER]) && !this._tableService.isFocusInsideTableCell)
+        ) {
+            this._tableService.processFocusInsideCell(this.navigationId);
+        }
+
+        const arrowKeyCodes = [UP_ARROW, DOWN_ARROW, LEFT_ARROW, RIGHT_ARROW];
+        if (KeyUtil.isKeyCode(event, arrowKeyCodes) && document.activeElement === this._elRef.nativeElement) {
```

Comment:
same as above, please swap the conditions

Reply:
Done